### ISSUE TITLE
Add support for bookmark events

### ISSUE THREAD

**Issue by cansjt** [kubernetes-client/python#1729]
**What is the feature and why do you need it**:

In kubernetes-client/python-base#234 the parsing of BOOKMARK events has been bypassed ([and still is as of today](https://github.com/kubernetes-client/python/blob/master/kubernetes/base/watch/watch.py#L103)). But those events exist for a reason and can be useful in some circumstances. A motivational example is this Airflow issue: https://github.com/apache/airflow/issues/21087.

As Airflows scheduler can be stopped and restarted, while many events occur on k8s, it would need to process BOOKMARK to recover its state. Sadly because those are ignored by the Python client, it cannot. Which manifests itself in varied ways (see the different issues referenced in the ticket above)

**Describe the solution you'd like to see**:

We'd like for the bookmark event to be parsed and yield by the watcher, so it is possible for clients to consume them.

(copied from kubernetes-client/python-base#286)

**Comment by roycaihw** [comment]
Agreed this is nice to have. 

/help

**Comment by snjypl** [comment]
@cansjt  thanks for reporting this.  i would like to make few comments: 


> **What is the feature and why do you need it**:
> 
> In [kubernetes-client/python-base#234](https://github.com/kubernetes-client/python-base/pull/234) the parsing of BOOKMARK events has been bypassed ([and still is as of today](https://github.com/kubernetes-client/python/blob/master/kubernetes/base/watch/watch.py#L103)). But those events exist for a reason and can be useful in some circumstances. A motivational example is this Airflow issue: [apache/airflow#21087](https://github.com/apache/airflow/issues/21087).
> 
The parsing of the BOOKMARK event is not actually by passed. we just  dont de-serialize the BOOKMARK event to the `return_type` eg: 'V1POD' etc.   the bookmark event as a dict is still returned by the watcher to the client. 

de serializing the event to the V1POD type etc is not easy because BOOKMARK event does not have all the required fields to do so. eg in case of V1POD, the container is None in BOOKMARK but required to be non empty for V1POD spec.

since all that really matter in the BOOKMARK event  in the resoureVersion, we should be good with a dict object. 

the problem that we have currently is the watcher does not update the resource_version to the one in the BOOKMARK event. which the #1796  should fix. 
once this fix is done, the 410 error won't happen for long duration watch. 

> As Airflows scheduler can be stopped and restarted, while many events occur on k8s, it would need to process BOOKMARK to recover its state. Sadly because those are ignored by the Python client, it cannot. Which manifests itself in varied ways (see the different issues referenced in the ticket above)
> 
The BOOKMARK event is not send by the api server by default. it needs to be requested by the client by setting the allow_watch_bookmarks  in the api call. by default it is false.  in Airflow kubernetes executor it is not set to True. 

so to take advantage of BOOKMARKS, in airflow  we need to make the request with `allow_watch_bookmark=true`
